ftd-host/auth.ftd

```diff
@@ -0,0 +1,121 @@
+-- import: fastn.com/ftd/built-in-variables as v
+-- import: fastn/processors as pr
+-- import: fastn.com/ftd-host/processor
+-- import: fastn.com/backend/env-vars
+-- import: fifthtry.github.io/admonitions as cbox
+
+-- ds.page: Github User Details using the `user-details` processor
+
+Let's look at how we can access basic user details of a user authenticated
+using GitHub.
+
+-- experimental-warning:
+-- processor.static-vs-dynamic:
+
+-- ds.markdown:
+
+We can use `user-details` processor to get information about the
+authenticated user. First, we need some data structures:
+
+-- ds.code: Types required by `user-details`
+lang: ftd
+
+\-- record status:
+boolean is-login:
+optional user-details user:
+
+\-- record user-details:
+string name:
+string login:
+integer id:
+
+-- ds.markdown:
+
+- `status` contains information about the user and a status representing
+  their login state.
+
+- `user-details` is the actual information that we get from GitHub. As of
+  now, you can only get the `login`(GitHub username), name, id (internal
+  GitHub user id).
+
+-- ds.h2: Using the `user-details` processor
+
+With the above data-structures defined, we can use them with the `user-details`
+processor:
+
+-- ds.code: `user-details` processor usage
+lang: ftd
+
+\-- import: fastn/processors as pr
+
+\-- status auth:
+\$processor$: pr.user-details
+
+-- ds.markdown:
+
+- The `auth` variable can now be used to access user details, below is a simple
+  example:
+
+-- ds.code: Example use in a webpage
+lang: ftd
+
+\-- ftd.column:
+
+\-- ftd.row:
+\if: { auth.is-login == false }
+
+\-- ftd.text: You're not logged in! Can't give you the details.
+
+\-- end: ftd.row
+
+\-- ftd.row:
+\if: { auth.is-login == true }
```

Comment:
Is this `== true` required?

Reply:
It is not. Will change this 👍

Reply:
For the `== false` case, do we have `!` (not operator)?

Reply:
yes we do. But I think `== false` is easier to read.